systick.c

```diff
@@ -0,0 +1,100 @@
+/*
+  Not part of Grbl. KeyMe specific.
+
+  SysTick implementation. Global SysTick value is updated.
+  Callbacks can be registed to be called after a certain time.
+
+  SysTick uses Timer1. Timer1 should not be used anywhere else.
+
+*/
+#include "systick.h"
+#include "nuts_bolts.h"
+
+#define MAX_CALLBACKS 32
+
+callback_t systick_callbacks_array[MAX_CALLBACKS];
+uint8_t systick_len;  // Length of callback array 
+
+void systick_service_callbacks()
+{
+  if (!systick_len) {
+    return;
+  }
+  // Assumes the callback array is sorted 
+  // according to lowest callback time
+  if (systick_callbacks_array[0].callback_time <= SysTick) {
```

Comment:
Thanks, that makes sense. 

1 uS was an example - I realize that it is unrealistic, but I am sure there are other more realistic cases with this flaw.

Alternatively, (as mentioned by you off-line) we can count the number of expired callbacks before the loop and only execute the callbacks which expired before starting the loop. By doing so, we avoid the infinite loop scenario, but still have the benefit of a harder real-time.